src/cli/commands/snip.ts

```diff
@@ -55,26 +76,30 @@ export const doRst = async (
   const rstEmphasizeModifier = ":emphasize-lines:";
 
   const rstEmphasizeRanges: { start: number; end: number }[] = [];
-  for (const range of emphasizeAttributes.ranges) {
-    const start = await document.getNewLocationFor(range.start);
-    const end = await document.getNewLocationFor(range.end);
-    if (start !== undefined && end !== undefined) {
-      rstEmphasizeRanges.push({ start: start.line, end: end.line });
+  let rstFormattedRanges = undefined;
```

Comment:
Minor: This section could be tidied up a bit. I would probably make this its own function (lines 80-102). (Actually, to be honest, I would do some hipster stuff with a chain of map() calls, but that's to taste...)

Reply:
Agreed, but I think I'll hold off on it at the moment because I think we should refactor it when we add *another* markup language, which will inform us a bit more about the common stages in the process and how to generalize this (though this PR has improved that). Next up: figuring out which language that should be!